beetsplug/parentwork.py

```diff
@@ -0,0 +1,159 @@
+
+"""Gets work title, disambiguation, parent work and its disambiguation,
+composer, composer sort name and performers
+"""
+
+from __future__ import division, absolute_import, print_function
+
+from beets import ui
+from beets.plugins import BeetsPlugin
+from beets.dbcore import types
+
+import musicbrainzngs
+
+
+class ParentWorkPlugin(BeetsPlugin):
+
+    def __init__(self):
+        super(ParentWorkPlugin, self).__init__()
+        self.import_stages = [self.imported]
+
+    def commands(self):
+        cmd = ui.Subcommand('parentwork',
+                            help=u'fetches parent works, composers \
+                                and performers')
+        cmd.func = self.command
+        return [cmd]
+
+    def command(self, lib, opts, args):
+        self.find_work(lib.items(ui.decargs(args)))
+
+    item_types = {
+        'parent_work':          types.STRING,
+        'parent_work_disambig': types.STRING,
+        'work':                 types.STRING,
+        'work_disambig':        types.STRING,
+        'performer':            types.STRING,
+        'performer_sort':       types.STRING,
+        'parent_composer':      types.STRING,
+        'parent_composer_sort': types.STRING}
+
+    def imported(self, session, task):
+        self.find_work(task.items)
+
+    def find_work(self, items):
+    
+        def find_parentwork(work_id):
+            "finds the parentwork of a work given its id"
+            work_info = musicbrainzngs.get_work_by_id(work_id,
+                includes=["work-rels", "artist-rels"])
+            partof = True
+            while partof:
+                partof = False
+                if 'work-relation-list' in work_info['work']:
+                    for work_father in work_info['work'][
+                            'work-relation-list']:
+                        if work_father['type'] == 'parts' and\
+                                'direction' in work_father:
+                            if work_father['direction'] == 'backward':
+                                father_id = work_father['work']['id']
+                                partof = True
+                                work_info = musicbrainzngs.get_work_by_id(
+                                    father_id,includes=["work-rels", 
+                                        "artist-rels"])
+            return work_info
+
+        for item in items:
+            performer            = []
+            performer_sort       = []
+            work                 = []
+            work_disambig        = []
+            parent_work          = []
+            parent_work_disambig = []
+            parent_composer      = []
+            parent_composer_sort = []
+            item.read()
+            recording_id = item['mb_trackid']
+            performer_types = ['performer', 'instrument', 'vocal',
+                'conductor', 'performing orchestra', 'chorus master', 
+                    'concertmaster']
+            i = 0
+            while i < 5:
```

Comment:
The idea is similar to the `fetchart` or `lyrics` plugins, where the plugin only actually does something if the music doesn't already have album art or lyrics. (There's an -f flag that overrides this behavior.) In general, this is a user-friendly way to make the process fast when there's only a little bit of new, unprocessed music.